common/ferrostar/src/models.rs

```diff
@@ -79,18 +77,16 @@ impl From<GeographicCoordinate> for Point {
 /// and are used for recalculating when the user deviates from the expected route.
 ///
 /// Note that support for properties beyond basic geographic coordinates varies by routing engine.
-#[derive(Clone, Copy, PartialEq, PartialOrd, Debug)]
+#[derive(Clone, Copy, PartialEq, PartialOrd, Debug, Serialize, Deserialize)]
 #[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
-#[cfg_attr(test, derive(Serialize))]
 pub struct Waypoint {
     pub coordinate: GeographicCoordinate,
     pub kind: WaypointKind,
 }
 
 /// Describes characteristics of the waypoint for the routing backend.
-#[derive(Clone, Copy, PartialEq, PartialOrd, Debug)]
+#[derive(Clone, Copy, PartialEq, PartialOrd, Debug, Serialize, Deserialize)]
```

Comment:
The reason I left these derives unconditioned is Rust doesn't allow duplicating traits, like:
```
#[cfg_attr(test, derive(Serialize))]
#[cfg_attr(feature = "wasm-bindgen", derive(Deserialize, Serialize))]
```
This will cause an error and i didn't want to make compiler conditions too huge, so I decided to let the compiler remove the unused codes.

However, I've found another approach to keep correct conditions and won't make a very long `cfg` block. Let's see if this will work.